ffi_interface/src/serialization.rs

```diff
@@ -131,6 +161,41 @@ pub fn deserialize_verifier_query(bytes: &[u8]) -> VerifierQuery {
     }
 }
 
+#[must_use]
+pub fn deserialize_verifier_query_uncompressed(bytes: &[u8]) -> VerifierQuery {
+    // Commitment
+    let (commitment, bytes) = take_uncompressed_group_element(bytes);
+
+    // The input point is a single byte
+    let (z_i, bytes) = take_byte(bytes);
+
+    // The evaluation is a single scalar
+    let (y_i, bytes) = take_scalar(bytes);
+
+    assert!(bytes.is_empty(), "we should have consumed all the bytes");
+
+    VerifierQuery {
+        commitment,
+        point: Fr::from(z_i as u128),
+        result: y_i,
+    }
+}
+
+#[must_use]
+pub(crate) fn take_uncompressed_group_element(bytes: &[u8]) -> (Element, &[u8]) {
+    let mut commitment: CommitmentBytes = bytes[..64]
+        .try_into()
+        .expect("Expected a slice of exactly 64 bytes");
+
+    commitment.reverse();
+    let (second_half, first_half) = commitment.split_at(32);
+    let commitment_reversed = [first_half, second_half].concat().try_into().unwrap();
+
+    let element = Element::from_bytes_unchecked_uncompressed(commitment_reversed);
```

Comment:
dont do this, just change how the serialization is implemented in c# to match how its done is rust